lib/interface-types/src/encoders/wat.rs

```diff
@@ -0,0 +1,716 @@
+                name = name,
+                inputs = input_types_to_param(&input_types),
+                outputs = output_types_to_result(&output_types),
+                instructions =
+                    instructions
+                        .iter()
+                        .fold(String::new(), |mut accumulator, instruction| {
+                            accumulator.push_str("\n  ");
+                            accumulator.push_str(&String::from(instruction));
+                            accumulator
+                        }),
+            ),
+
+            _ => unimplemented!(),
+        }
+    }
+}
+
+/// Encode a `Forward` into a string.
+impl<'input> From<&Forward<'input>> for String {
+    fn from(forward: &Forward) -> Self {
+        format!(
+            r#"(@interface forward (export "{name}"))"#,
+            name = forward.name,
+        )
+    }
+}
+
+/// Encode an `Interfaces` into a string.
+impl<'input> From<&Interfaces<'input>> for String {
+    fn from(interfaces: &Interfaces) -> Self {
+        let mut output = String::from(";; Interfaces");
+
+        let exports = interfaces
+            .exports
+            .iter()
+            .fold(String::new(), |mut accumulator, export| {
+                accumulator.push_str(&format!("\n\n;; Interface, Export {}\n", export.name));
+                accumulator.push_str(&String::from(export));
+                accumulator
+            });
+
+        let types = interfaces
+            .types
+            .iter()
+            .fold(String::new(), |mut accumulator, ty| {
+                accumulator.push_str(&format!("\n\n;; Interface, Ty {}\n", ty.name));
+                accumulator.push_str(&String::from(ty));
+                accumulator
+            });
+
+        let imports = interfaces
+            .imports
+            .iter()
+            .fold(String::new(), |mut accumulator, import| {
+                accumulator.push_str(&format!(
+                    "\n\n;; Interface, Import {}.{}\n",
+                    import.namespace, import.name
+                ));
+                accumulator.push_str(&String::from(import));
+                accumulator
+            });
+
+        let adapters =
+            interfaces
+                .adapters
+                .iter()
+                .fold(String::new(), |mut accumulator, adapter| {
+                    match adapter {
+                        Adapter::Import {
+                            namespace, name, ..
+                        } => accumulator.push_str(&format!(
+                            "\n\n;; Interface, Adapter {}.{}\n",
+                            namespace, name
+                        )),
+
+                        Adapter::Export { name, .. } => {
+                            accumulator.push_str(&format!("\n\n;; Interface, Adapter {}\n", name))
+                        }
+
+                        _ => unimplemented!(),
```

Comment:
Here too, any string is fine really, but ideally it should be identifiable and as helpful as it can be

Reply:
Done!